rueidiscompat/command.go

```diff
@@ -509,44 +510,49 @@ func newStatusCmd(res rueidis.RedisResult) *StatusCmd {
 }
 
 type SliceCmd struct {
-	err  error
-	val  []any
+	baseCmd[[]any]
+	// val  []any
 	keys []string
 }
 
-func newSliceCmd(res rueidis.RedisResult, keys ...string) *SliceCmd {
-	val, err := res.ToArray()
-	slice := &SliceCmd{val: make([]any, len(val)), err: err, keys: keys}
-	for i, v := range val {
-		if s, err := v.ToString(); err == nil {
-			slice.val[i] = s
+// newSliceCmd returns SliceCmd according to input arguments, if the caller is JSONObjKeys,
+// set isJSONObjKeys to true.
+func newSliceCmd(res rueidis.RedisResult, isJSONObjKeys bool, keys ...string) *SliceCmd {
+	cmd := &SliceCmd{keys: keys}
+	arr, err := res.ToArray()
+	if err != nil {
+		cmd.SetErr(err)
+		return cmd
+	}
+	vals := make([]any, len(arr))
+	for i, v := range arr {
+		if isJSONObjKeys {
+			// for JSON.OBJKEYS
+			if v.IsNil() {
+				vals[i] = nil
```

Comment:
This is unnecessary, right?

Reply:
It's necessary because the return value might be `Nil`

e.g. 

```
redis> JSON.SET doc $ '{"a":[3], "nested": {"a": {"b":2, "c": 1}}}'
OK
redis> JSON.OBJKEYS doc $..a
1) (nil)
2) 1) "b"
   2) "c"
```

Ref: [`JSON.OBJKEYS`](https://redis.io/commands/json.objkeys/)

Reply:
I get your point, `vals[i]` is `nil` by default, no need to set it, I'll fix that.